packages/cli/src/UserManagement/controllers/UserController.ts

```diff
@@ -0,0 +1,485 @@
+/* eslint-disable import/no-cycle */
+import { Response } from 'express';
+import { In } from 'typeorm';
+import validator from 'validator';
+import { LoggerProxy as Logger } from 'n8n-workflow';
+import type { UserRequest } from '../../requests';
+import { Delete, Get, Post, RestController } from '../decorators';
+import { ActiveWorkflowRunner, Db, ITelemetryUserDeletionData, ResponseHelper } from '../..';
+import { User } from '../../databases/entities/User';
+import { SharedWorkflow } from '../../databases/entities/SharedWorkflow';
+import { SharedCredentials } from '../../databases/entities/SharedCredentials';
+import { InternalHooksManager } from '../../InternalHooksManager';
+import * as UserManagementMailer from '../email';
+import {
+	isUserManagementDisabled,
+	getInstanceBaseUrl,
+	validatePassword,
+	hashPassword,
+	sanitizeUser,
+	isEmailSetUp,
+} from '../UserManagementHelper';
+import * as config from '../../../config';
+import { issueCookie } from '../auth/jwt';
+
+@RestController('/users')
+export class UserController {
+	/**
+	 * Send email invite(s) to one or multiple users and create user shell(s).
+	 */
+	@Post()
+	async sendEmailInvites(req: UserRequest.Invite) {
+		if (config.getEnv('userManagement.emails.mode') === '') {
+			Logger.debug(
+				'Request to send email invite(s) to user(s) failed because emailing was not set up',
+			);
+			throw new ResponseHelper.ResponseError(
+				'Email sending must be set up in order to request a password reset email',
+				undefined,
+				500,
+			);
+		}
+
+		let mailer: UserManagementMailer.UserManagementMailer | undefined;
+		try {
+			mailer = await UserManagementMailer.getInstance();
+		} catch (error) {
+			if (error instanceof Error) {
+				throw new ResponseHelper.ResponseError(
+					`There is a problem with your SMTP setup! ${error.message}`,
+					undefined,
+					500,
+				);
+			}
+		}
+
+		// TODO: this should be checked in the middleware rather than here
+		if (isUserManagementDisabled()) {
+			Logger.debug(
+				'Request to send email invite(s) to user(s) failed because user management is disabled',
+			);
+			throw new ResponseHelper.ResponseError('User management is disabled');
+		}
+
+		if (!config.getEnv('userManagement.isInstanceOwnerSetUp')) {
+			Logger.debug(
+				'Request to send email invite(s) to user(s) failed because the owner account is not set up',
+			);
+			throw new ResponseHelper.ResponseError(
+				'You must set up your own account before inviting others',
+				undefined,
+				400,
+			);
+		}
```

Comment:
Good candidates for future decorator guards: `@userManagementEnabled`, `@instanceOwnerSetUp`, `@emailModeSmtp`

Reply:
agreed